e2e/test/DeviceStreamingE2ETests.cs

```diff
@@ -415,8 +415,10 @@ private async Task TestEchoThroughStreamingGatewayAsync(ClientDeviceStreamingReq
                 }, TaskScheduler.Current)
             ).ConfigureAwait(false);
 
-            await deviceWSClient.CloseAsync(WebSocketCloseStatus.NormalClosure, "End of test", cts.Token).ConfigureAwait(false);
-            await serviceWSClient.CloseAsync(WebSocketCloseStatus.NormalClosure, "End of test", cts.Token).ConfigureAwait(false);
+            Task.WaitAll(
```

Comment:
Mixing Wait and Await is extremely dangerous and can cause deadlocks.
Use `await Task.WhenAll(...).ConfigureAwait(false);` instead.

Reply:
Thanks Cris. Shall be changed.